### PR TITLE
Upgrade Azure SDK version

### PR DESCRIPTION
**Pull request by josegrobles**
On this PR we have upgraded the used Azure SDK version to the latest one (7.0.0). 
As we have had to jump through so many versions, some changes were required but the API is stable so nothing should change on the clients using this library.
We have also improved the tests even though their functionality remains unchanged.

### CODE REVIEW

**Review by reymon359 — COMMENTED** [review]
Looks great! 👍🏽 

Btw I had an overview and think that maybe some of the useful comments from @jgleal for the migration are not necessary anymore like `// JGL - no sense. It seems to work fine now` [here](https://github.com/guidesmiths/systemic-azure-bus/blob/d1e105266f8e6f74aafaf67c5689beab5f0d5a5b/test/topics/dlq.test.js#L60) 🤔 

Just a reminder that we should remove them to not have them on `master` if we don't need them 😊

**Comment by josegrobles** [comment]
> Looks great! 👍🏽
> 
> Btw I had an overview and think that maybe some of the useful comments from @jgleal for the migration are not necessary anymore like `// JGL - no sense. It seems to work fine now` [here](https://github.com/guidesmiths/systemic-azure-bus/blob/d1e105266f8e6f74aafaf67c5689beab5f0d5a5b/test/topics/dlq.test.js#L60) 🤔
> 
> Just a reminder that we should remove them to not have them on `master` if we don't need them 😊

Yeah, I'll remove them :)

**Comment by josegrobles** [comment]
Also for some reason Travis CI doesn't want to pass :(

File: test/topics/dlq.test.js [hunk 1]
```diff
@@ -67,6 +73,7 @@ describe('Topics - Systemic Azure Bus API', () => {
 		schedule(peekDlq);
 	}));
 
+	// JGL - with this test, retry suite fails U_u
```
Comment (jgleal) [inline test/topics/dlq.test.js:76]: remove it 😇

**Review by jgleal — APPROVED** [review]

**Pull request merged**